### PR TITLE
Fix OProfile discovery and drop bfd dependency

### PR DESCRIPTION
**Pull request by ShadowsFriend**
The current implementation did not find the OProfile library. This adds a new cmake module to find OProfile and adjusts the CMakeLists.txt to take advantage of it.

### CODE REVIEW

**Comment by phire** [comment]
Well, I guess all of the *nix buildbots are broken.

File: CMakeTests/FindOProfile.cmake [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# - Try to find OProfile
+# Once done this will define
+# OPROFILE_FOUND - System has OProfile
+# OPROFILE_INCLUDE_DIRS - The OProfile include directories
+# OPROFILE_LIBRARIES - The libraries needed to use OProfile
+
+if(CMAKE_SIZEOF_VOID_P EQUAL 8)
+    set(_lib_suffix 64)
```
Comment (phire) [inline CMakeTests/FindOProfile.cmake]: Shouldn't /lib/ always be a simlink to the correct /lib64 or /lib32 depending on the system?

Are there any systems where this is not true?

**Comment by ShadowsFriend** [comment]
I honestly don't know. The check shouldn't hurt though, should it?

**Comment by phire** [comment]
I guess not. Just seems hacky to test the size of `void*` There are a few ABIs that have 32bit pointers despite being 64bit, like x32 or the ps3 (not that dolphin will ever be running on either of those platforms)

**Comment by ShadowsFriend** [comment]
I actually copied that part from obs-studio's FindZLIB.cmake because this was recommended to me as an example and it seemed reasonable to me. I can remove that part if its too hacky for you, especially considering that Dolphin is officially announced to only run on 64 bit systems.

**Comment by Sonicadvance1** [comment]
We also detect 32bit or 64bit architectures via that check in the main CMakeLists file.
There is a comment in there explaining that it'll break on x32 ABI systems and that if we start supporting that architecture then it will need to be changed.
It's currently the best way to determine what the bitness of the architecture is without adding other checks.

**Comment by phire** [comment]
We don't need to detect the bitness of the system. /lib should always point to your library, if it doesn't, you are cross compiling, and you have bigger problems and /lib64 isn't the correct place either.

The person who wrote this .cmake file was just paranoid.

**Comment by ShadowsFriend** [comment]
I just removed the bfd dependency from the CMakelists for the OProfile option because it is not needed at all. While OProfile itself depends on it, Dolphin doesn't. And there is no reason to link against a library not needed.

**Comment by phire** [comment]
@dolphin-emu-bot rebuild

**Comment by ShadowsFriend** [comment]
I squashed the three commits into one and changed the title of the PR to better reflect the current proposal.

File: CMakeTests/FindOProfile.cmake [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+# OPROFILE_FOUND - System has OProfile
+# OPROFILE_INCLUDE_DIRS - The OProfile include directories
+# OPROFILE_LIBRARIES - The libraries needed to use OProfile
+
+set(FIND_OPROFILE_PATHS
+    /usr/local
+    /usr
+    /sw
+    /opt/local
+    /opt)
```
Comment (Tilka) [inline CMakeTests/FindOProfile.cmake]: Is this really necessary?

File: CMakeTests/FindOProfile.cmake [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+find_path(OPROFILE_INCLUDE_DIR opagent.h
+    PATHS ${FIND_OPROFILE_PATHS}
+    PATH_SUFFIXES include)
+
+find_library(OPROFILE_LIB
```
Comment (Tilka) [inline CMakeTests/FindOProfile.cmake]: Please use `OPROFILE_LIBRARY`.

File: CMakeTests/FindOProfile.cmake [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    NAMES opagent libopagent
```
Comment (Tilka) [inline CMakeTests/FindOProfile.cmake]: CMake automatically adds the `lib` prefix.

File: CMakeTests/FindOProfile.cmake [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    PATHS ${FIND_OPROFILE_PATHS}
+    PATH_SUFFIXES
+	lib
+	lib/oprofile
+	oprofile)
```
Comment (Tilka) [inline CMakeTests/FindOProfile.cmake]: I don't think `lib` and `lib/oprofile` are necessary.

File: CMakeTests/FindOProfile.cmake [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+set(OPROFILE_INCLUDE_DIRS ${OPROFILE_INCLUDE_DIR})
+set(OPROFILE_LIBRARIES ${OPROFILE_LIBRARY})
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(OProfile DEFAULT_MSG
+    OPROFILE_LIB OPROFILE_INCLUDE_DIR)
+
+mark_as_advanced(OPROFILE_INCLUDE_DIR OPROFILE_LIB)
```
Comment (Tilka) [inline CMakeTests/FindOProfile.cmake]: Same as above.

File: CMakeTests/FindOProfile.cmake (hunk 2 above)
Comment (ShadowsFriend) [inline CMakeTests/FindOProfile.cmake]: For a standard installation of OProfile probably not. Should I drop everything besides /usr and /usr/local?

Comment (BhaaLseN) [inline CMakeTests/FindOProfile.cmake]: How does this behave on OSX, if at all? IIRC, port puts its stuff somewhere under /opt

Comment (Tilka) [inline CMakeTests/FindOProfile.cmake]: I assume it's CMake's job to know that and look there.

**Comment by ShadowsFriend** [comment]
Stripped it down to the everything that is needed to find OProfile in /usr as well as in /usr/local. And implemented the requested changes. Thanks for the feedback and sorry for the mess.

**Comment by Tilka** [comment]
lgtm

**Pull request merged**